siad/api_test.go

```diff
@@ -0,0 +1,340 @@
+	}
+
+	// Create the http request
+	req, err := http.NewRequest("POST", "http://127.0.0.1:9980/file/upload", body)
+	if err != nil {
+		t.Fatal("Error creating POST request: " + err.Error())
+	}
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	// Actually make the request
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal("Error creating http client: " + err.Error())
+	}
+
+	// Read response body
+	resContent, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Fatal("Error reading response content: " + err.Error())
+	}
+
+	// Check the response
+	if res.StatusCode != http.StatusOK {
+		t.Fatal("/file/upload returne " + strconv.Itoa(res.StatusCode) + ": " + string(resContent))
+	}
+
+	//TODO: Read response into JSON success object
+	//TODO: return success object
+}
+
+func reqDownloadFile(t *testing.T, nickname string, filename string) SuccessResponse {
+	return reqSuccess(t, "/file/download?nickname="+nickname+"&filename="+filename)
+}
+
+func reqStop(t *testing.T) {
+	httpReq(t, "/stop")
+}
+
+func setupDaemon(t *testing.T) {
+
+	// Settings to speed up operations
+	consensus.BlockFrequency = consensus.Timestamp(1)
+	consensus.TargetWindow = consensus.BlockHeight(1000)
+	network.BootstrapPeers = []network.Address{"localhost:9988"}
+	consensus.RootTarget[0] = 255
+	consensus.MaxAdjustmentUp = big.NewRat(1005, 1000)
+	consensus.MaxAdjustmentDown = big.NewRat(995, 1000)
+
+	var config Config
+
+	config.Siad.ConfigFilename = filepath.Join(siaDir, "config")
+	config.Siacore.HostDirectory = filepath.Join(siaDir, "hostdir")
+	config.Siad.StyleDirectory = filepath.Join(siaDir, "style")
+	config.Siad.DownloadDirectory = "~/Downloads"
+	config.Siad.WalletFile = filepath.Join(siaDir, "test.wallet")
+	config.Siad.APIaddr = "localhost:9980"
+	config.Siacore.RPCaddr = ":9988"
+	config.Siacore.NoBootstrap = false
+	err := config.expand()
+	if err != nil {
+		t.Fatal("Couldn't expand config: " + err.Error())
+	}
+
+	go func() {
+		err = startDaemon(config)
+		if err != nil {
+			t.Fatal("Couldn't start daemon: " + err.Error())
+		}
+	}()
+
+	// Give the daemon time to initialize
+	time.Sleep(10 * time.Millisecond)
+
+	// First call is just to see if daemon booted
+	_, err = httpReq(t, "/wallet/status")
+	if err != nil {
+		t.Fatal("Daemon could not handle first request (after 10ms) " + err.Error())
+	}
+}
+
+// Tests that the initial status of the daemon is sane - additionally tests that
+// each component can return it's status
+func TestBasicStatus(t *testing.T) {
+	setupDaemon(t)
+
+	// Test getting miner status
+	mInfo := reqMinerStatus(t)
+	if mInfo.State != "Off" {
+		t.Fatal("Miner initial state was not off: (" + strconv.Itoa(mInfo.Threads) + ")")
+	}
+	if mInfo.RunningThreads != 0 {
+		t.Fatal("Miner initial RunningThreads was not 0: (" + strconv.Itoa(mInfo.Threads) + ")")
+	}
+	if mInfo.Threads != 1 {
+		t.Fatal("Miner initial Threads was not 1 (" + strconv.Itoa(mInfo.Threads) + ")")
+	}
```

Comment:
I'm not sure how necessary these tests are, because it's not necessarily behavior that we want to enforce. Starting in the "Off" state is good, and running threads is probably okay too. So I guess it's just the last one.